src/sentry/migrations/0685_alert_rule_project_backfill_migration.py

```diff
@@ -0,0 +1,66 @@
+# Generated by Django 5.0.3 on 2024-03-22 22:36
+
+import logging
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+logger = logging.getLogger(__name__)
+
+
+def _backfill_alert_rule_projects(apps, schema_editor):
+    QuerySubscriptions = apps.get_model("sentry", "QuerySubscription")
+    AlertRuleProjects = apps.get_model("sentry", "AlertRuleProjects")
+
+    for subscription in QuerySubscriptions.objects.all():
+
+        snuba_query = subscription.snuba_query
+        if not snuba_query:
+            logger.warning(
+                "QuerySubscription found with no snuba_query",
+                extra={"query_subscription_id": subscription.id},
+            )
+            continue
+
+        alert_rule = snuba_query.alertrule_set.get()
+
+        existing_alert_rule_projects = AlertRuleProjects.objects.filter(alert_rule=alert_rule)
```

Comment:
It might make sense to just coerce this to a list instead and operate on the model instances, since you're performing a lot of different operations here